### ISSUE TITLE
building error with torch

### ISSUE THREAD

**Issue by shuaitang** [dmlc/mxnet#1254]
There is no problem when I compile mxnet with USE_TORCH=0.

But when I set USE_TORCH=1, the problem turns out like this

~/torch/install/lib/libTH.so: undefined reference to `sgeqrf_'
~/torch/install/lib/libTH.so: undefined reference to `ssyev_'
~/torch/install/lib/libTH.so: undefined reference to `sgesvd_'
~/torch/install/lib/libTH.so: undefined reference to `dsyev_'
~/torch/install/lib/libTH.so: undefined reference to `dorgqr_'
~/torch/install/lib/libTH.so: undefined reference to `dpotri_'
~/torch/install/lib/libTH.so: undefined reference to `dgesv_'
~/torch/install/lib/libTH.so: undefined reference to `sgels_'
~/torch/install/lib/libTH.so: undefined reference to `dgels_'
~/torch/install/lib/libTH.so: undefined reference to `dtrtrs_'
~/torch/install/lib/libTH.so: undefined reference to `sorgqr_'
~/torch/install/lib/libTH.so: undefined reference to `spotrs_'
~/torch/install/lib/libTH.so: undefined reference to `dpstrf_'
~/torch/install/lib/libTH.so: undefined reference to `sgetri_'
~/torch/install/lib/libTH.so: undefined reference to `dpotrf_'
~/torch/install/lib/libTH.so: undefined reference to `dgetri_'
~/torch/install/lib/libTH.so: undefined reference to `sormqr_'
~/torch/install/lib/libTH.so: undefined reference to `dgeqrf_'
~/torch/install/lib/libTH.so: undefined reference to `strtrs_'
~/torch/install/lib/libTH.so: undefined reference to `dormqr_'
~/torch/install/lib/libTH.so: undefined reference to `dpotrs_'
~/torch/install/lib/libTH.so: undefined reference to `sgesv_'
~/torch/install/lib/libTH.so: undefined reference to `dgetrf_'
~/torch/install/lib/libTH.so: undefined reference to `dgeev_'
~/torch/install/lib/libTH.so: undefined reference to `sgetrf_'
~/torch/install/lib/libTH.so: undefined reference to `dgesvd_'
~/torch/install/lib/libTH.so: undefined reference to `spotrf_'
~/torch/install/lib/libTH.so: undefined reference to `spstrf_'
~torch/install/lib/libTH.so: undefined reference to `sgeev_'
~/torch/install/lib/libTH.so: undefined reference to `spotri_'
collect2: error: ld returned 1 exit status
make: *** [bin/im2rec] Error 1

Is it caused by torch or mxnet?

Thank you

**Comment by antinucleon** [comment]
It seems like torch link problem (with altas or lapack). Check your torch building.

**Comment by piiswrong** [comment]
does your torch installation work?

**Comment by shuaitang** [comment]
yes, torch works well, and openblas is ok...

but the problem is here...

**Comment by piiswrong** [comment]
try adding -L/path/to/lapack -llapack?

**Comment by shuaitang** [comment]
# the additional link flags you want to add
ADD_LDFLAGS =-L/usr/lib/lapack -llapack

i add it like this, but it still doesnt work

**Comment by sodabeta7** [comment]
@shuaitang  Have you fixed the problem? I've also find the exactly same problem and can not figure out the solution.

**Comment by Conchylicultor** [comment]
Same compilation error with a clean torch install on ubuntu 14.04, gcc 4.8 . Did someone solve the issue ?

**Comment by TheOncomingStorm** [comment]
I experienced the same kind of error yesterday and the problem ended up being that I'd compiled lapack with gcc 4.x but had compiled torch with 5.x. The "undefined reference to..." is a huge hint that's what's going on as I've been slowly upgrading my (Gentoo) system and I've hit that error countless times. That may not be your problem but I figured I'd post a potential solution in case it helps someone.
